.github/workflows/packages.yml

```diff
@@ -295,3 +295,71 @@ jobs:
 #            -e ICINGA_BUILD_DEB_DEFAULT_ARCH=armhf \
 #            registry.icinga.com/build-docker/raspbian/${{ matrix.codename }} \
 #            icinga-build-test
+  windows:
+    name: Windows
+
+    strategy:
+      fail-fast: false
+      matrix:
+        bits: [32, 64]
+
+    runs-on: windows-latest
+
+    env:
+      BITS: '${{ matrix.bits }}'
+      ICINGA_BUILD_TYPE: snapshot
+      UPSTREAM_GIT_URL: file://D:/a/icinga2/icinga2/.git
+
+    steps:
+      - name: Checkout HEAD
+        uses: actions/checkout@v1
+
+      - name: windows-icinga2
+        run: |
+          git clone https://git.icinga.com/packaging/windows-icinga2.git
+
+      - name: Build tools
+        run: |
+          Set-PSDebug -Trace 1
+          function Install-Exe {
+            param (
+              [string]$Url,
+              [string]$Dir
+            )
+            $TempDir = Join-Path ([System.IO.Path]::GetTempPath()) ([System.Guid]::NewGuid().Guid)
+            $ExeFile = Join-Path $TempDir inst.exe
+            New-Item -ItemType Directory -Path $TempDir
+            Invoke-WebRequest -Uri $Url -OutFile $ExeFile -UseBasicParsing
+            Start-Process -Wait -FilePath $ExeFile -ArgumentList @('/VERYSILENT', '/INSTALL', '/PASSIVE', '/NORESTART', "/DIR=${Dir}")
+            Remove-Item -Recurse -Path $TempDir
+          }
+          $VsVersion = 2019
+          $BoostVersion = @(1, 71, 0)
+          $OpensslVersion = '1_1_1h'
+          # VS
+          choco install -y "visualstudio${VsVersion}community"
+          choco install -y "visualstudio${VsVersion}-workload-netcoretools"
+          choco install -y "visualstudio${VsVersion}-workload-vctools"
+          choco install -y "visualstudio${VsVersion}-workload-manageddesktop"
+          choco install -y "visualstudio${VsVersion}-workload-nativedesktop"
+          choco install -y "visualstudio${VsVersion}-workload-universal"
+          choco install -y "visualstudio${VsVersion}buildtools"
+          # Misc
+          choco install -y cmake
+          choco install -y winflexbison3
+          choco install -y windows-sdk-8.1
+          choco install -y wixtoolset
+          Install-Exe -Url "https://master.dl.sourceforge.net/project/boost/boost-binaries/$($BoostVersion -join '.')/boost_$($BoostVersion -join '_')-msvc-14.2-${Env:BITS}.exe" -Dir "C:\local\boost_$($BoostVersion -join '_')-Win${Env:BITS}"
+          Install-Exe -Url "https://slproweb.com/download/Win${Env:BITS}OpenSSL-${OpensslVersion}.exe" -Dir "C:\local\OpenSSL_${OpensslVersion}-Win${Env:BITS}"
```

Comment:
Is there a way to use custom images with this pre-installed? Should give shorter build times and make the mirrors happier, if this isn't downloaded for every build. Or is it at least possible to cache files between runs?

Reply:
1. They've VS _Enterprise_, but I'd like to have the same software as in GitLab CI.
2. Caching? Technically – sure. Storage-ly – not really. Legally – I don't think so.

Reply:
I don't get what storage and legal problems you see. What I was thinking about was "Is it possible to define `C:\local` as a cached directory that is kept between executions?" similar to what can be done in GitLab CI: https://docs.gitlab.com/ce/ci/caching/

Reply:
But then we'd cache only Boost and OpenSSL. Would it make a difference?

Reply:
The biggest chunk is probably VS, which can't be cached easily I guess. So only a small difference. But if that just 2 lines of config (I don't know if it actually is), I'd say why not?

Reply:
Those aren't _just 2 lines of config_. We'd have to also check whether the cache is present in PS.

Reply:
I don't have any strong opinions here. As it's only on Windows, it's just 2 jobs anyways, things would be different if it was Linux and this would be downloaded 20 times for every pipeline.